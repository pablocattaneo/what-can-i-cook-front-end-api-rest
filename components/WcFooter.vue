<template>
  <div id="wc-footer" class="mt-5 bg-dark container-fluid">
    <div class="row">
      <div class="col-12 d-flex justify-content-center mt-3">
        <ul class="list-unstyled text-center">
          <li class="mt-2">
            <b-img-lazy
              blank
              blank-color="#abc"
              width="24"
              src="@/assets/img/linkedin-icon.svg"
              alt="placeholder"
            />
            <a
              id="stackoverflow-user-profile-url"
              v-if="linkedinProfileUrl"
              :href="linkedinProfileUrl"
              class="mt-0 mb-1"
              target="_blank"
              >Linkdin</a
            >
          </li>
          <li class="mt-2">
            <b-img-lazy
              blank
              blank-color="#abc"
              width="24"
              src="@/assets/img/stack-overflow-icon.svg"
              alt="placeholder"
            />
            <a
              id="stackoverflow-user-profile-url"
              v-if="stackoverflowUserProfileUrl"
              :href="stackoverflowUserProfileUrl"
              class="mt-0 mb-1"
              target="_blank"
              >Stackoverflow</a
            >
          </li>
        </ul>
      </div>
    </div>
    <div class="text-center row author-section">
      <div class="col-12">
        <span id="author"
          >{{ $t('App_deloped_by') }}
          <a
            v-if="author.externalLink"
            :href="author.externalLink"
            target="blank"
            >{{ author.name }}</a
          >
          <span v-else>{{ author.name }}</span>
        </span>
      </div>
    </div>
    <div class="row icon-made">
      <div class="col-12">
        <p class="text-center">
          Icons made by
          <a
            href="https://www.flaticon.com/free-icon/chef_2413394"
            title="iconixar"
            >iconixar</a
          >
          from
          <a href="https://www.flaticon.com/" title="Flaticon"
            >www.flaticon.com</a
          >
        </p>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    author: {
      type: Object,
      default: () => ({
        name: '',
        externalLink: ''
      })
    },
    stackoverflowUserProfileUrl: {
      type: String,
      default: ''
    },
    linkedinProfileUrl: {
      type: String,
      default: ''
    }
  }
}
</script>

<style lang="stylus" scoped>
#wc-footer
  color white
  .icon-made,
  .author-section
    p,
    a,
    span
      font-size 15px
      @media (min-width: $md)
        font-size 12px
</style>
