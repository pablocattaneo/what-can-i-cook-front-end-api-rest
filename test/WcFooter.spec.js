import { mount, createLocalVue } from '@vue/test-utils'
import WcFooter from '@/components/WcFooter'
import { BootstrapVue } from 'bootstrap-vue'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

let wrapper
beforeAll(() => {
  wrapper = mount(WcFooter, {
    localVue
  })
})

describe('Author Property', () => {
  test('Author name property was set to Pablo Cattaneo so component should render this text.', async () => {
    await wrapper.setProps({ author: { name: 'Pablo Cattaneo' } })
    expect(wrapper.find('#author').text()).toContain('Pablo Cattaneo')
  })
  test('Author externalLink was set to https://www.linkedin.com/in/cattaneopablo/ so component should render this a element with href with this value.', async () => {
    await wrapper.setProps({
      author: { externalLink: 'https://www.linkedin.com/in/cattaneopablo/' }
    })
    expect(wrapper.find('#author a').attributes().href).toBe(
      'https://www.linkedin.com/in/cattaneopablo/'
    )
  })
  test('Author externalLink was falsy so component should render span element instead a link.', async () => {
    await wrapper.setProps({
      author: { externalLink: '' }
    })
    expect(wrapper.find('#author a').exists()).toBe(false)
    expect(wrapper.find('#author span').exists()).toBe(true)
  })
})

describe('StackOverflow link', () => {
  test('stackoverflowUserProfileUrl props was set to https://stackoverflow.com/users/3599272/pablo so component should be rendered.', async () => {
    await wrapper.setProps({
      stackoverflowUserProfileUrl:
        'https://stackoverflow.com/users/3599272/pablo'
    })
    expect(wrapper.find('a#stackoverflow-user-profile-url').exists()).toBe(true)
  })
  test('stackoverflowUserProfileUrl props was set to https://stackoverflow.com/users/3599272/pablo so component should a link elemnt whith this value in this href property.', async () => {
    await wrapper.setProps({
      stackoverflowUserProfileUrl:
        'https://stackoverflow.com/users/3599272/pablo'
    })
    expect(
      wrapper.find('a#stackoverflow-user-profile-url').attributes().href
    ).toBe('https://stackoverflow.com/users/3599272/pablo')
  })
})
