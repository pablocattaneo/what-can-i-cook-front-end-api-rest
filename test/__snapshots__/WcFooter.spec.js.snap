// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots It verify if non-intentional changes were made. 1`] = `
<div
  class="mt-5 bg-dark container-fluid"
  id="wc-footer"
>
  <div
    class="row"
  >
    <div
      class="col-12 d-flex justify-content-center mt-3"
    >
      <ul
        class="list-unstyled text-center"
      >
        <li
          class="mt-2"
        >
          <b-img-lazy-stub
            alt="placeholder"
            blank=""
            blankcolor="#abc"
            offset="360"
            src="@/assets/img/linkedin-icon.svg"
            width="24"
          />
           
          <!---->
        </li>
         
        <li
          class="mt-2"
        >
          <b-img-lazy-stub
            alt="placeholder"
            blank=""
            blankcolor="#abc"
            offset="360"
            src="@/assets/img/stack-overflow-icon.svg"
            width="24"
          />
           
          <!---->
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="text-center row author-section"
  >
    <div
      class="col-12"
    >
      <span
        id="author"
      >
        
        
        <span>
          
        </span>
      </span>
    </div>
  </div>
   
  <div
    class="row icon-made"
  >
    <div
      class="col-12"
    >
      <p
        class="text-center"
      >
        
        Icons made by
        
        <a
          href="https://www.flaticon.com/free-icon/chef_2413394"
          title="iconixar"
        >
          iconixar
        </a>
        
        from
        
        <a
          href="https://www.flaticon.com/"
          title="Flaticon"
        >
          www.flaticon.com
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`linkedinProfileUrl link Branch: linkedinProfileUrl props was set to https://www.linkedin.com/in/cattaneopablo/ so it verify if non-intentional changes were made for this output. 1`] = `
<div
  class="mt-5 bg-dark container-fluid"
  id="wc-footer"
>
  <div
    class="row"
  >
    <div
      class="col-12 d-flex justify-content-center mt-3"
    >
      <ul
        class="list-unstyled text-center"
      >
        <li
          class="mt-2"
        >
          <b-img-lazy-stub
            alt="placeholder"
            blank=""
            blankcolor="#abc"
            offset="360"
            src="@/assets/img/linkedin-icon.svg"
            width="24"
          />
           
          <a
            class="mt-0 mb-1"
            href="https://www.linkedin.com/in/cattaneopablo/"
            id="linkdin-url"
            target="_blank"
          >
            Linkdin
          </a>
        </li>
         
        <li
          class="mt-2"
        >
          <b-img-lazy-stub
            alt="placeholder"
            blank=""
            blankcolor="#abc"
            offset="360"
            src="@/assets/img/stack-overflow-icon.svg"
            width="24"
          />
           
          <!---->
        </li>
      </ul>
    </div>
  </div>
   
  <div
    class="text-center row author-section"
  >
    <div
      class="col-12"
    >
      <span
        id="author"
      >
        
        
        <span>
          
        </span>
      </span>
    </div>
  </div>
   
  <div
    class="row icon-made"
  >
    <div
      class="col-12"
    >
      <p
        class="text-center"
      >
        
        Icons made by
        
        <a
          href="https://www.flaticon.com/free-icon/chef_2413394"
          title="iconixar"
        >
          iconixar
        </a>
        
        from
        
        <a
          href="https://www.flaticon.com/"
          title="Flaticon"
        >
          www.flaticon.com
        </a>
      </p>
    </div>
  </div>
</div>
`;
